angular/src/lib/pagination/test/pagination.component.spec.ts

```diff
@@ -3,106 +3,205 @@ import { PaginationService } from '../pagination.service';
-    expect(numberComponentsText).toContain('1 2 3 4 5');
-  });
-
-  it('should correctly render 6 buttons', () => {
-    component.total = 6;
-    component.current = 5;
-    component.ngOnInit();
-    fixture.detectChanges();
+@Component({
+  template: `
+    <md-pagination
+      [total]='totalPages'
+      [current]='currentPage'
+      [firstGroupNum]='firstGroupNum'
+      [midGroupNum]='midGroupNum'
+      [lastGroupNum]='lastGroupNum'
+    ></md-pagination>
+  `
+})
+class PaginationWrapperComponent {
+  totalPages: number;
+  currentPage: number;
+  firstGroupNum: number = 2;
+  midGroupNum: number = 3;
+  lastGroupNum: number = 2;
+}
 
-    const ellipsis: DebugElement[] = fixture.debugElement.queryAll(By.css('li.pagination_li.ellipsis'));
-    expect(ellipsis.length).toBe(0);
-
-    const numberComponents: DebugElement[] = fixture.debugElement.queryAll(By.directive(PaginationNumberComponent));
-    const numberComponentsText = numberComponents
-      .reduce((accumulator, currentElement) => accumulator + ' ' + currentElement.nativeElement.textContent, '');
-    expect(numberComponentsText).toContain('1 2 3 4 5 6');
+describe('PaginationComponent', () => {
+  describe('creation and rendering', () => {
+    let component: PaginationComponent;
+    let fixture: ComponentFixture<PaginationComponent>;
+    let testElement: HTMLElement;
+    let testElements: Array<HTMLElement>;
+    let paginationService: PaginationService;
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        declarations: [ PaginationComponent ],
+        imports: [PaginationArrowModule, PaginationNumberModule],
+        providers: [ PaginationService ]
+      })
+      .compileComponents();
+    }));
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(PaginationComponent);
+      component = fixture.componentInstance;
+      fixture.detectChanges();
+    });
+
+    it('should create and match snapshot', () => {
+      expect(component).toBeTruthy();
+      expect(fixture).toMatchSnapshot();
+    });
+
+    it('should create', () => {
+      expect(component).toBeTruthy();
+    });
+
+    it('should run service Overlay', () => {
+      component.ngOnInit();
+      fixture.detectChanges();
+      paginationService = TestBed.get(PaginationService);
+      expect(paginationService).toBeTruthy();
+    });
+
+    it('should render total 3', () => {
+      component.total = 3;
+      component.current = 2;
+      component.ngOnInit();
+      fixture.detectChanges();
+      testElements = fixture.nativeElement.getElementsByTagName('a');
+      expect(testElements.length).toEqual(4);
+    });
+
+    it('should disable previous', () => {
+      component.total = 3;
+      component.ngOnInit();
+      fixture.detectChanges();
+      testElement = fixture.nativeElement.getElementsByTagName('md-pagination-arrow')[0];
+      expect(testElement.className).toContain('icon-arrow-left-optical_20');
```

Comment:
how does this test that the previous button is disabled?  It is only looking for the left arrow which I expect would be there either way